### ISSUE TITLE
BUG: read_excel with engine=openpyxl is including nanoseconds (inaccurate at times it seems) in datetimes where xlrd did not

### ISSUE THREAD

**Issue by jtgutt** [pandas-dev/pandas#39599]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
>>> import pandas as pd
>>> pd.read_excel('~/testDatetimeNanos.xlsx')
             TestDate
0 2020-09-24 01:00:00
1 2020-09-09 16:26:00
2 2020-09-09 16:26:00
>>> pd.read_excel('~/testDatetimeNanos.xlsx', engine='openpyxl')
                    TestDate
0 2020-09-24 01:00:00.000003
1 2020-09-09 16:25:59.999998
2 2020-09-09 16:26:00.000000
```
[testDatetimeNanos.xlsx](https://github.com/pandas-dev/pandas/files/5928342/testDatetimeNanos.xlsx)

#### Problem description
_Problem also described in [this Stack Overflow post](https://stackoverflow.com/questions/66037692/how-to-make-pandas-read-excel-with-engine-openpyxl-behave-like-it-did-with-xlr)._

The default behavior of the read_excel function seems to have changed when using the openpyxl engine from how it used to behave when using the xlrd engine.  It looks like now datetimes are being represented with nanoseconds and in some cases appear to differ by ~2-3 nanoseconds from the expected datetime value as seen in Excel.

#### Expected Output
The expected output is reflected in the first dataframe shown above, where the datetimes do not include nanoseconds, as this has been the behavior previously for xlsx files using xlrd.
 
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.6.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.9.230-0.1.ac.223.84.332.metal1.x86_64
Version          : #1 SMP Tue Sep 8 15:48:58 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.4
numpy            : 1.19.4
pytz             : 2020.1
dateutil         : 2.8.0
pip              : 21.0.1
setuptools       : 43.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : None
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.6
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : 0.9.6
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None

</details>

**Comment by nmartensen** [comment]
Time is stored as a simple numeric value in xlsx, with limited precision. With xlrd, the returned datetimes are rounded to the nearest millisecond. With openpyxl, there is no rounding (except around date boundaries), so you can see the limited precision in the microseconds. I have a patch to openpyxl changing this that I plan to submit soon.

**Comment by nmartensen** [comment]
This is fixed in openpyxl 3.0.7, which now also returns datetimes rounded to the nearest millisecond.

**Comment by mroeschke** [comment]
Closing as this appears to be an upstream issue that has been fixed
